modules/yadm/mod.nu

```diff
@@ -1,1672 +1,4236 @@
+            unset-all:$unset_all 
+            remove-section:$remove_section
+            rename-section:$rename_section
+            comment:$comment
+          }
+      )
+    
+  }
+  
+  
+  export module git-crypt  { 
+    
+
+    # Share the repository using GPG
+    export def add-gpg-user [user_id:string] {
+      
+      bash_yadm git-crypt add-gpg-user $user_id
+
+    }
+    
+    # Configure a repository to use git-crypt
+    export def init [] {
+      
+      bash_yadm git-crypt init
+
+    }
+    
+    # Unlock encrypted files
+    export def unlock [] {
+      
+      bash_yadm git-crypt unlock
+
+    }
+
+    # Export a symmetric secret key
+    export def export-key [key_file_path:string] {
+      
+      bash_yadm git-crypt export-key $key_file_path
+ 
+    }
+    
+  }
+  
+  export def transcrypt [
+    --cipher(-c):string 
+
+    --password(-p):string 
+
+    --set-openssl-path:string 
+
+    --yes(-y) 
+
+    --display(-d) 
+
+    --rekey(-r) 
+
+    --flush-credentials(-f) 
+
+    --force(-F) 
+
+    --uninstall(-u) 
+
+    --upgrade 
+
+    --list(-l) 
+
+    --show-raw(-s):string 
+
+    --export-gpg(-e):string 
+
+    --import-gpg(-i) 
+
+    --context(-C) 
+
+    --list-contexts 
+
+    --version(-v) 
+
+    --help(-h) 
+  ] { 
+    
+  bash_yadm transcrypt (
+      generate_viable_bash_string_flags {
+        cipher:$cipher
+        password:$password
+        set-openssl-path:$set_openssl_path
+        yes:$yes
+        display:$display
+        rekey:$rekey
+        flush-credentials:$flush_credentials
+        force:$force
+        uninstall:$uninstall
+        upgrade:$upgrade
+        show-raw:$show_raw
+        export-gpg:$export_gpg
+        import-gpg:$import_gpg
+        context:$context
+        list-contexts:$list_contexts
+        version:$version
+        help:$help
+      }
+    )
+  
+  }
+  
+  export def alt [file_path:string] { bash_yadm alt $file_path }
+  
+  export def encrypt [] { bash_yadm encrypt }
+  
+  export def decrypt [] { bash_yadm decrypt }
+  
+  export def perms [file_path:string] { bash_yadm perms $file_path }
+  
+  export def bootstrap [] { bash_yadm bootstrap }
+  
+  export def enter [] { bash_yadm enter }
+  
+  export module introspect { 
+    const possible_subjects = [
+      'commands'
+      'repos'
+      'switches'
+      'configs'
+    ]
+
+    def get_completions_for_subject [] {
+        $possible_subjects
+    }
+    
+    export def main [subject:string@get_completions_for_subject, ] {
+      
+      if($subject  not-in $possible_subjects) {
+
+        let joined_possible_subjects = $possible_subjects 
+        | str join ","
+
+        error make {
+          msg: "Invalid Argument",
+          label: "This value is incorrect!"
+          span:(metadata $subject).span
+          help: $"This value ($value) isn't correct the ones are ($joined_possible_subjects). " 
+        }
+      }
+
+      bash_yadm introspect $subject
+      
+    }
+  
+  }
+  
+module git {

```

Comment:
I kind of wonder if having a module named `git` will stomp on other modules that people use already named `git`. Seems like this one could be `yadm_git` or some such thing to disambiguate.

Reply:
The user is supposed to use `export module yadm` to use this module.
If the user decides to use `*` there nothing I could do. 
The git module is there for me to group commands related to git without repetition.

Reply:
but doesn't that module get exposed once you use yadm? it's a real question, i haven't tried it to know for sure. i just saw the export at the end of the file and was hoping it didn't overwrite or conflict with our other git modules in this repo.

Reply:
No, I don't export the module, just its commands.